### ISSUE TITLE
RJDBC unable to handle numeric(18,2)

### ISSUE THREAD

**Issue by perNyfelt** [bedatadriven/renjin#443]
Using RJDBC on PostgreSQL i get an error when reading from a table containing a column of type numeric(18,2) as follows:

R code was:
library('org.renjin.cran:RJDBC')
library(DBI)
 
drv <- JDBC("org.postgresql.Driver") 
con <- dbConnect(drv, 
url="jdbc:postgresql://localhost:5432/rsg", user="xxx", password="xxx") 

projectInfoId <- 218
components <-  dbGetQuery(con, paste("select * from component where project_id = ", projectInfoId))

Error:
org.renjin.eval.EvalException: Unknown column type list(name = unitcost, type = numeric)

I think the numeric type should map to BigDecimal (or double if BigDecimal is a lot of work). 

org.renjin.eval.EvalException: Unknown column type list(name = unitcost, type = numeric)
	at org.renjin.invoke.reflection.FunctionBinding$Overload.invoke(FunctionBinding.java:125)
	at org.renjin.invoke.reflection.FunctionBinding.invoke(FunctionBinding.java:172)
	at org.renjin.invoke.reflection.FunctionBinding.evaluateArgsAndInvoke(FunctionBinding.java:154)
	at org.renjin.invoke.reflection.MethodFunction.apply(MethodFunction.java:58)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:38)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.eval.Context.evaluate(Context.java:198)
	at org.renjin.sexp.Closure.doApply(Closure.java:98)
	at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:80)
	at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:50)
	at org.renjin.sexp.Closure.apply(Closure.java:83)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:38)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.eval.Context.evaluate(Context.java:198)
	at org.renjin.sexp.Closure.doApply(Closure.java:98)
	at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:80)
	at org.renjin.methods.Methods.standardGeneric(Methods.java:568)
	at org.renjin.primitives.R$primitive$standardGeneric.doApply(R$primitive$standardGeneric.java:79)
	at org.renjin.primitives.R$primitive$standardGeneric.apply(R$primitive$standardGeneric.java:33)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.primitives.special.AssignLeftFunction.assignLeft(AssignLeftFunction.java:57)
	at org.renjin.primitives.special.AssignLeftFunction.apply(AssignLeftFunction.java:41)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:38)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.eval.Context.evaluate(Context.java:198)
	at org.renjin.sexp.Closure.doApply(Closure.java:98)
	at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:80)
	at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:50)
	at org.renjin.sexp.Closure.apply(Closure.java:83)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:38)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.eval.Context.evaluate(Context.java:198)
	at org.renjin.sexp.Closure.doApply(Closure.java:98)
	at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:80)
	at org.renjin.methods.Methods.standardGeneric(Methods.java:568)
	at org.renjin.primitives.R$primitive$standardGeneric.doApply(R$primitive$standardGeneric.java:79)
	at org.renjin.primitives.R$primitive$standardGeneric.apply(R$primitive$standardGeneric.java:33)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.eval.Context.evaluate(Context.java:198)
	at org.renjin.sexp.Closure.doApply(Closure.java:98)
	at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:80)
	at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:50)
	at org.renjin.sexp.Closure.apply(Closure.java:83)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.primitives.special.AssignLeftFunction.assignLeft(AssignLeftFunction.java:57)
	at org.renjin.primitives.special.AssignLeftFunction.apply(AssignLeftFunction.java:41)
	at org.renjin.eval.Context.evaluateCall(Context.java:407)
	at org.renjin.eval.Context.evaluate(Context.java:282)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.eval.Context.evaluateExpressionVector(Context.java:390)
	at org.renjin.eval.Context.evaluate(Context.java:280)
	at org.renjin.eval.Context.evaluate(Context.java:264)
	at org.renjin.script.RenjinScriptEngine.eval(RenjinScriptEngine.java:168)
	at org.renjin.script.RenjinScriptEngine.eval(RenjinScriptEngine.java:127)

**Comment by perNyfelt** [comment]
Looking at the "caused by" part of the stacktrace it seems that JDBCUtils is used to map values. 
Caused by: org.renjin.eval.EvalException: Unknown column type list(name = current_loan_amount, type = numeric)
	at org.renjin.cran.DBI.JDBCUtils.fetch(JDBCUtils.java:185)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.renjin.invoke.reflection.FunctionBinding$Overload.invoke(FunctionBinding.java:117)

I am not sure how this ties into RJDBC but a simple fix for the issue would be to add numeric to the double type i.e: in org.renjin.JDBC.DoubleColumnBinder change the method acceptsType to include numeric as follows:

    public static boolean acceptsType(String columnType) {
      return columnType.equals("decimal") || columnType.equals("real") || columnType.equals("number") || columnType.equals("numeric")
          || columnType.startsWith("double") || columnType.startsWith("float");
    }

**Comment by perNyfelt** [comment]
I figured out that RJDBC is part of renjin-dbi. I suggest a change to the classes DateStringColumnBuilder (add support for the timestamp type) and DoubleColumnBuilder (add support for the numeric type)
 The code can be seen in https://github.com/perNyfelt/renjin-dbi/tree/master/RJDBC/src/main/java/org/renjin/cran/DBI/columns

there is also a hamcrest java test here: https://github.com/perNyfelt/renjin-dbi/blob/master/RJDBC/src/test/R/TestH2.R. A h2 test dependency needs to be added to the pom for the test to work i.e:

		<dependency>
			<groupId>com.h2database</groupId>
			<artifactId>h2</artifactId>
			<version>1.4.197</version>
			<scope>test</scope>
		</dependency>

As I could not  quite the the renjin plugin to work so I used my own hamcrest test plugin I cannot create a pull request but hope that this should be helpful anyway.
